command/pki_verify_sign_command.go

```diff
@@ -0,0 +1,231 @@
+		for _, arg := range args {
+			if strings.HasPrefix(arg, "-") {
+				c.UI.Warn(fmt.Sprintf("Options (%v) must be specified before positional arguments (%v)", arg, args[0]))
+				break
+			}
+		}
+		return 1
+	}
+
+	issuer := sanitizePath(args[0])
+	issued := sanitizePath(args[1])
+
+	client, err := c.Client()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Failed to obtain client: %s", err))
+		return 1
+	}
+
+	err, results := verifySignBetween(client, issuer, issued)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Failed to run verification: %v", err))
+		return pkiRetUsage
+	}
+
+	c.outputResults(results, issuer, issued)
+
+	return 0
+}
+
+func verifySignBetween(client *api.Client, issuerPath string, issuedPath string) (error, map[string]bool) {
+	// Note that this eats warnings
+
+	// Fetch and Parse the Potential Issuer:
+	issuerResp, err := client.Logical().Read(issuerPath)
+	if err != nil {
+		return fmt.Errorf("error: unable to fetch issuer %v: %w", issuerPath, err), nil
+	}
+	issuerCertPem := issuerResp.Data["certificate"].(string)
+	issuerCertBundle, err := certutil.ParsePEMBundle(issuerCertPem)
+	if err != nil {
+		return err, nil
+	}
+	issuerKeyId := issuerCertBundle.Certificate.SubjectKeyId
+
+	// Fetch and Parse the Potential Issued Cert
+	issuedCertResp, err := client.Logical().Read(issuedPath)
+	if err != nil {
+		return fmt.Errorf("error: unable to fetch issuer %v: %w", issuerPath, err), nil
+	}
+	if len(issuedPath) <= 2 {
+		return fmt.Errorf(fmt.Sprintf("%v", issuedPath)), nil
+	}
+	caChainRaw := issuedCertResp.Data["ca_chain"]
+	if caChainRaw == nil {
+		return fmt.Errorf("no ca_chain information on %v", issuedPath), nil
+	}
+	caChainCast := caChainRaw.([]interface{})
+	caChain := make([]string, len(caChainCast))
+	for i, cert := range caChainCast {
+		caChain[i] = cert.(string)
+	}
+	issuedCertPem := issuedCertResp.Data["certificate"].(string)
+	issuedCertBundle, err := certutil.ParsePEMBundle(issuedCertPem)
+	if err != nil {
+		return err, nil
+	}
+	parentKeyId := issuedCertBundle.Certificate.AuthorityKeyId
+
+	// Check the Chain-Match
+	rootCertPool := x509.NewCertPool()
+	rootCertPool.AddCert(issuerCertBundle.Certificate)
+	checkTrustPathOptions := x509.VerifyOptions{
+		Roots: rootCertPool,
+	}
+	trust := false
+	trusts, err := issuedCertBundle.Certificate.Verify(checkTrustPathOptions)
+	if err != nil && !strings.Contains(err.Error(), "certificate signed by unknown authority") {
+		return err, nil
+	} else if err == nil {
+		for _, chain := range trusts {
+			// Output of this Should Only Have One Trust with Chain of Length Two (Child followed by Parent)
+			for _, cert := range chain {
+				if issuedCertBundle.Certificate.Equal(cert) {
+					trust = true
+					break
+				}
+			}
+		}
+	}
+
+	pathMatch := false
+	for _, cert := range caChain {
+		if strings.TrimSpace(cert) == strings.TrimSpace(issuerCertPem) { // TODO: Decode into ASN1 to Check
```

Comment:
nit: Was this a future TODO or does this need to be addressed sooner? Similar for the TODO below on line 177